sources/kubernetes/kubernetes-proxy/src/main.rs

```
@@ -0,0 +1,173 @@
+        + for<'de> serde::Deserialize<'de>
+        + std::fmt::Debug
+        + serde::Serialize
+        + 'static,
+{
+    if req.node_labels.contains(&K::KIND.to_string()) {
+        if let Err(e) = tokio::spawn(stream_resources::<K>(kube_client.clone(), tx.clone())).await {
+            log::error!("Error streaming {}: {}", K::KIND, e);
+            return Err(BootstrapError::InternalError(e.to_string()));
+        };
+    }
+    Ok(())
+}
+
+async fn stream_resources<K>(
+    kube_client: Client,
+    tx: UnboundedSender<SourceElement>,
+) -> Result<(), BootstrapError>
+where
+    <K as kube::Resource>::DynamicType: Default,
+    K: kube::Resource
+        + Metadata
+        + std::clone::Clone
+        + Send
+        + Sync
+        + for<'de> serde::Deserialize<'de>
+        + std::fmt::Debug
+        + serde::Serialize
+        + 'static,
+{
+    let page_size = 50;
+    let api = Api::<K>::all(kube_client.clone());
+
+    let mut items = match api.list(&ListParams::default().limit(page_size)).await {
+        Ok(items) => items,
+        Err(e) => return Err(BootstrapError::InternalError(e.to_string())),
+    };
+
+    loop {
+        for p in items.items {
+            let id = match p.uid() {
+                Some(id) => id,
+                None => continue,
+            };
+
+            for r in p.owner_references() {
+                _ = tx.send(SourceElement::Relation {
+                    id: format!("{}-{}", r.uid, id),
+                    labels: vec!["OWNS".to_string()],
+                    properties: Map::new(),
+                    start_id: r.uid.clone(),
+                    end_id: id.clone(),
+                });
+            }
+
+            let properties = match serde_json::to_value(p) {
+                Ok(Value::Object(map)) => map,
+                _ => continue,
+            };
+
+            let label = match properties.get("kind") {
+                Some(Value::String(kind)) => kind.clone(),
+                _ => continue,
```

If `kind` is not a string, should we throw an error? Also in this case, a relation will be created but the item node will not be created. Will this be an issue?

---

If `kind` was not a string, then something would be very broken with Kubernetes... perhaps we should error out.